diff --git a/node_modules/metro-file-map/src/watchers/FallbackWatcher.js b/node_modules/metro-file-map/src/watchers/FallbackWatcher.js
index 75ac8a0..0a69b92 100644
--- a/node_modules/metro-file-map/src/watchers/FallbackWatcher.js
+++ b/node_modules/metro-file-map/src/watchers/FallbackWatcher.js
@@ -85,13 +85,22 @@ module.exports = class FallbackWatcher extends AbstractWatcher {
     if (this.watched[dir]) {
       return false;
     }
-    const watcher = fs.watch(
-      dir,
-      {
-        persistent: true,
-      },
-      (event, filename) => this._normalizeChange(dir, event, filename)
-    );
+    if (!fs.existsSync(dir)) {
+      return false;
+    }
+    let watcher;
+    try {
+      watcher = fs.watch(
+        dir,
+        {
+          persistent: true,
+        },
+        (event, filename) => this._normalizeChange(dir, event, filename)
+      );
+    } catch (error) {
+      this._checkedEmitError(error);
+      return false;
+    }
     this.watched[dir] = watcher;
     watcher.on("error", this._checkedEmitError);
     if (this.root !== dir) {
